performance_manager/lib/rt_trip_updates.py

```diff
@@ -0,0 +1,250 @@
+from typing import Optional, Union, List, Dict, Any
+
+import logging
+import pandas
+import numpy
+
+import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+
+from .s3_utils import read_parquet
+from .gtfs_utils import start_time_to_seconds, add_event_hash_column
+from .postgres_utils import TripUpdateEvents, MetadataLog, get_unprocessed_files
+
+
+def get_tu_dataframe(to_load: Union[str, List[str]]) -> pandas.DataFrame:
+    """
+    return a dataframe from a trip updates parquet file (or list of files)
+    with expected columns
+    """
+    trip_update_columns = [
+        "timestamp",
+        "stop_time_update",
+        "direction_id",
+        "route_id",
+        "start_date",
+        "start_time",
+        "vehicle_id",
+    ]
+    trip_update_filters = [
+        ("direction_id", "in", (0, 1)),
+        ("timestamp", ">", 0),
+        ("route_id", "!=", "None"),
+        ("start_date", "!=", "None"),
+        ("start_time", "!=", "None"),
+        ("vehicle_id", "!=", "None"),
+    ]
+
+    return read_parquet(
+        to_load, columns=trip_update_columns, filters=trip_update_filters
+    )
+
+
+def explode_stop_time_update(row: pandas.Series) -> Optional[List[dict]]:
+    """
+    function to be used with pandas `apply` dataframe method
+    explode nested list of dicts in stop_time_update column
+    """
```

Comment:
This the function that is "applied" to every record on our dataframe from the parquet files. This is what results in high-ish memory usage. We'll have to keep an eye on resources in the ECS, if they become an issue, this is the most likely culprit.

Reply:
is there a way that we can log the memory usage we're getting after applying this explode func?

Reply:
Our best bet would probably be to use python's `subprocess` library and grep `free` output???

Reply:
We would also have to thread the `apply` call and whatever method we have for logging the memory usage, since they would have to run concurrently.